src/bms/command/time.rs

```diff
@@ -19,16 +21,32 @@ pub struct ObjTime {
 
 impl ObjTime {
     /// Create a new time.
-    ///
-    /// # Panics
-    ///
-    /// Panics if `denominator` is 0 or `numerator` is greater than or equal to `denominator`.
-    pub fn new(track: u64, numerator: u64, denominator: u64) -> Self {
-        if track == 0 {
-            eprintln!("warning: track 000 detected");
+    pub fn new(mut track: u64, mut numerator: u64, mut denominator: u64) -> Self {
+        // If denominator is 0, set numerator to 0 and denominator to 1, and return.
+        if denominator == 0 {
+            return Self {
+                track: Track(track),
+                numerator: 0,
+                denominator: 1,
+            };
+        }
```

Comment:
The reason why it panics when `denominator == 0` is setting `denominator` zero is to prevent from mistaking about programming. In most cases, `ObjTime::new` is called with track index, index in the id sequence, and length of the id sequence. So I thought it should be discourageed; to pass `0` as `denominator` lacks check about the application logic.